bin/VMSlogger

```diff
@@ -120,57 +179,95 @@ class Collector:
-
-                async def collect_it():
-                    with click.progressbar(
-                        length=self.chunk_size,
-                        label=f"Getting data {devices[self.index]}",
-                        show_eta=True,
-                        show_percent=True,
-                        width=0,
-                    ) as bar:
-                        current = 0
-                        while not (self.cache.filled):
-                            l = len(self.cache)
-                            bar.update(l - current)
-                            current = l
-                            await asyncio.sleep(0.1)
-                            if self._save_hdf5():
-                                break
-                        bar.update(len(self.cache) - current)
+                await asyncio.sleep(0.5)
+        else:
 
+            async def collect_it(bar):
+                last_l = 0
+
+                while True:
+                    cache_len = len(self.cache)
+                    if cache_len >= self.size:
+                        break
+                    bar.update(cache_len - last_l)
+                    last_l = cache_len
+                    await asyncio.sleep(0.1)
+                    if self._save_hdf5():
+                        bar.update(self.chunk_size - last_l)
+                        break
+
+            with click.progressbar(
+                length=self.size,
+                label=f"Getting data {devices[self.index]}",
+                show_eta=True,
+                show_percent=True,
+                width=0,
+            ) as bar:
                 if self.h5file is None:
-                    await collect_it()
+                    await collect_it(bar)
                 else:
                     while True:
-                        await collect_it()
+                        await collect_it(bar)
                         if self.cache.h5_filled():
                             break
 
-            if self.h5file is None:
-                logger.info(f"Saving CVS to {self.filename}")
-                kwargs = {"delimiter": ","}
-                if self.header:
-                    kwargs["header"] = ",".join(self.cache.columns())
-                self.cache.savetxt(self.filename, **kwargs)
+                bar.update(self.size)
 
+        if self.h5file is None:
+            logger.info(f"Saving CSV to {self.filename}")
+            kwargs = {"delimiter": ","}
+            if self.header:
+                kwargs["header"] = ",".join(self.cache.columns())
+            self.cache.savetxt(self.filename, self.size, **kwargs)
+
+    async def collect_data(self):
+        try:
+            async with Domain() as domain:
+                remote = Remote(domain, "MTVMS", index=self.index + 1)
+                remote.tel_data.callback = lambda data: self.cache.newChunk(data, 0.001)
+                while True:
+                    self._create_file()
+                    await self.sample_file()
```

Comment:
How to you stop this `while` loop? By `cancel`?

Reply:
Yes, by killing the process. But I added --single-shot option to quit after one file (with -s records) is recorded.